common/tmx_filetype.go

```diff
@@ -0,0 +1,62 @@
+package common
+
+import (
+	"bytes"
+	"engo.io/engo"
+	"fmt"
+	"io"
+)
+
+// TMXResource contains a level created from a Tile Map XML
+type TMXResource struct {
+	Level *Level
+	url   string
+}
+
+// URL retrieves the url to the .tmx file
+func (r TMXResource) URL() string {
+	return r.url
+}
+
+// tmxLoader is responsible for managing '.tmx' files within 'engo.Files'.
+// You can generate a TMX file with the Tiled map editor.
+type tmxLoader struct {
+	levels map[string]TMXResource
+}
+
+// Load will load the tmx file and any other image resources that are needed
+func (t *tmxLoader) Load(url string, data io.Reader) error {
+	buf := new(bytes.Buffer)
+	_, err := buf.ReadFrom(data)
+	if err != nil {
+		return err
+	}
+
+	lvl, err := createLevelFromTmx(buf.String(), url)
```

Comment:
You're converting `[]byte` to `string` here, allocating new memory. 

However, when using it, you're casting it from `string` to `[]byte` again. You should simply pass along the `[]byte`.